Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_baking.py

```diff
@@ -0,0 +1,1195 @@
+            channelSettings = bakeSettings.bake_channel_settings.add()
+            channelSettings.name = channel
+
+
+class MESHSYNC_PT_Baking(MESHSYNC_PT, bpy.types.Panel):
+    bl_label = "Baking"
+    bl_parent_id = "MESHSYNC_PT_Main"
+    bl_order = -10
+
+    def draw(self, context):
+        layout = self.layout
+        layout.use_property_split = True
+        layout.use_property_decorate = False
+
+        bakeSettings = context.scene.meshsync_bake_settings
+
+        layout.prop(bakeSettings, "bake_selection", expand=True)
+
+        box = layout.box()
+        box.alignment = 'LEFT'
+        box.label(text="Material channels to bake")
+        box.prop(bakeSettings, "bake_all_channels")
+        for channelSetting in bakeSettings.bake_channel_settings:
+            row = box.row()
+            row.prop(channelSetting, "bakeChannelEnabled", text=channelSetting.name)
+
+        layout.prop(bakeSettings, "generate_uvs", expand=True)
+        layout.prop(bakeSettings, "apply_modifiers")
+        layout.prop(bakeSettings, "run_modal")
+        layout.operator("meshsync.bake_materials")
+
+        if bakeSettings.bake_progress > 0.0:
+            box = layout.box()
+            box.prop(bakeSettings, "bake_progress")
+            box.label(text=bakeSettings.bake_message, icon='INFO')
+            box.label(text=bakeSettings.bake_maps_remaining)
+            if len(bakeSettings.bake_time_remaining) > 0:
+                box.label(text=bakeSettings.bake_time_remaining)
+
+        row = layout.row()
+        row.prop(bakeSettings, "bakedTexturesPath")
+        row.operator("meshsync.choose_material_bake_folder", icon="FILE_FOLDER", text="")
+        layout.prop(bakeSettings, "bakedTextureSize")
+        layout.operator("meshsync.revert_bake_materials")
+
+
+class MESHSYNC_OT_Bake(bpy.types.Operator):
+    bl_idname = "meshsync.bake_materials"
+    bl_label = "Bake to individual materials"
+    bl_description = "Bakes textures, creates material copies and assigns the baked materials for all materials that " \
+                     "cannot be exported without baking them to textures"
+
+    maxBakeProgress = 0
+    currentBakeProgress = 0
+
+    def isMaterialCopy(self, mat):
+        return ORIGINAL_MATERIAL in mat
+
+    def canMaterialBeBaked(self, mat):
+        return mat is not None and \
+            not self.isMaterialCopy(mat) and \
+            mat.use_nodes
+
+    def findMaterialOutputNode(self, node_tree):
+        outputNode = None
+        for node in node_tree.nodes:
+            if node.type == 'GROUP':
+                outputNodeInGroup = self.findMaterialOutputNode(node.node_tree)
+                if outputNodeInGroup is not None:
+                    outputNode = outputNodeInGroup
+            elif node.type == 'OUTPUT_MATERIAL' and len(node.inputs[0].links) == 1:
+                outputNode = node
+                # Blender uses the last OUTPUT_MATERIAL node, so don't stop search here
```

Comment:
If you have two Material Output nodes and switch which one you're using by changing the link, will it definitely re-order the list to set the Material Output node you're using as the last Material Output node?

Reply:
I've improved that part of the code now. There's a better way to check that.